Add should process support for remove d365database

FH-Inway opened the pull request:
This pull request introduces several enhancements to the `Remove-D365Database` function and its usage in the `New-D365Bacpac` function. The key changes include adding new parameters to improve user control and updating documentation and tests accordingly.

fixes #879 

Enhancements to `Remove-D365Database` function:

* Added `-Confirm`, `-WhatIf`, and `-Force` parameters to `Remove-D365Database` to provide better control over command execution and user confirmation. (`d365fo.tools/functions/remove-d365database.ps1`, [[1]](diffhunk://#diff-f382fa87c0825e392abcfabbd8cbd4f86a31bd6ee4e3cb6ad919ce2c81bd1b9fR29-R37) [[2]](diffhunk://#diff-f382fa87c0825e392abcfabbd8cbd4f86a31bd6ee4e3cb6ad919ce2c81bd1b9fL40-R49) [[3]](diffhunk://#diff-f382fa87c0825e392abcfabbd8cbd4f86a31bd6ee4e3cb6ad919ce2c81bd1b9fL51-R67)
* Updated the function to support `ShouldProcess` for high-impact actions, ensuring users are prompted for confirmation before executing critical steps. (`d365fo.tools/functions/remove-d365database.ps1`, [d365fo.tools/functions/remove-d365database.ps1R89-R99](diffhunk://#diff-f382fa87c0825e392abcfabbd8cbd4f86a31bd6ee4e3cb6ad919ce2c81bd1b9fR89-R99))

Updates to `New-D365Bacpac` function:

* Modified calls to `Remove-D365Database` to include `-Confirm:$false`, bypassing confirmation prompts during automated processes. (`d365fo.tools/functions/new-d365bacpac.ps1`, [[1]](diffhunk://#diff-204f9a5102fb7c36b566342f12192ec69c016e1132595c4481d80a1520c9bebfL299-R299) [[2]](diffhunk://#diff-204f9a5102fb7c36b566342f12192ec69c016e1132595c4481d80a1520c9bebfL340-R340)

Documentation updates:

* Updated `Remove-D365Database` documentation to include descriptions and usage examples for the new `-Confirm`, `-WhatIf`, and `-Force` parameters. (`docs/Remove-D365Database.md`, [[1]](diffhunk://#diff-db5488ea16be060e14c1cdc4994a727b54068c5c296ccf7f0c4dc9a46dbfadc1L17-R17) [[2]](diffhunk://#diff-db5488ea16be060e14c1cdc4994a727b54068c5c296ccf7f0c4dc9a46dbfadc1R115-R161)

Testing enhancements:

* Added tests to validate the presence and behavior of the new `-Force` parameter in `Remove-D365Database`. (`d365fo.tools/tests/functions/Remove-D365Database.Tests.ps1`, [d365fo.tools/tests/functions/Remove-D365Database.Tests.ps1R79-R97](diffhunk://#diff-3be80bc9fb8c9dee94cad9e890707ced1b5784d0307a1e861b3e23ae73fdc5caR79-R97))

FH-Inway commented:
@Splaxi This will break Remove-D365Database for existing users that do not expect a confirmation prompt when running the command. 

My guess is not many are using this command and its main use is as part of New-D365Bacpac, where the confirmation prompt is suppressed.

One alternative I can think of (well, besides just not making the change) is to lower the `ConfirmImpact` to `Medium`. This would result in no prompt unless someone sets the `$ConfirmationPreference` to `Medium` (default is `High`). This would probably not have prevented #879 even if advise is added to the ReadMe to run d365fo.tools with Medium ConfirmationPreference while one gets familiar with it.

Splaxi commented:
I would like to go beyond and above to avoid breaking changes.

So let's handle the level and test things

FH-Inway commented:
Alright, I changed the impact level to Medium. In turn, I added some documentation and console messages to make the impact of the command clearer.
Also fixed some minor issues with the command.

New behavior looks like this:

> PS C:\Repositories\GitHub\FH-Inway\d365fo.tools> $OriginalConfirmPreference = $ConfirmPreference
PS C:\Repositories\GitHub\FH-Inway\d365fo.tools> $ConfirmPreference = "Medium"
PS C:\Repositories\GitHub\FH-Inway\d365fo.tools> Remove-D365Database -DatabaseName "AxDW"
> 
> Confirm
Are you sure you want to perform this action?
Performing the operation "Kill all processes" on target "AxDW".
[Y] Yes  [A] Yes to All  [N] No  [L] No to All  [S] Suspend  [?] Help (default is "Y"): L
PS C:\Repositories\GitHub\FH-Inway\d365fo.tools> $ConfirmPreference = $OriginalConfirmPreference
PS C:\Repositories\GitHub\FH-Inway\d365fo.tools> Remove-D365Database -DatabaseName "AxDW"
[14:11:33][Remove-D365Database] Database AxDW was removed.
PS C:\Repositories\GitHub\FH-Inway\d365fo.tools> Remove-D365Database
[14:11:51][Remove-D365Database] Database AxDB was removed.
PS C:\Repositories\GitHub\FH-Inway\d365fo.tools> Remove-D365Database
WARNING: [14:17:18][Remove-D365Database] Database AxDB not found. Nothing to remove.

![image](https://github.com/user-attachments/assets/709762ae-08a8-43cd-b278-7d591898281e)
